File: sdk/python/src/ks_app/canonical.py
"""canonical 派生：与 ks-types Canonical(appID, name) 等价。

两语言 SDK 各自实现同一派生逻辑（Go 用 kstypes.Canonical，Python 一份），
靠 sdk/shared-fixtures wire-compat 锁结果一致。
"""
from __future__ import annotations


def canonical(app_id: str, name: str) -> str:
    """由 app_id + 裸名派生全局唯一 canonical_name。"""
    return f"{app_id}.{name}"

File: sdk/typescript/src/canonical.ts
/**
 * canonical 派生：与 Go kstypes.Canonical / Python canonical 等价。
 * 三语言各自实现，靠 sdk/shared-fixtures/canonical_derivation.json wire-compat 锁一致。
 *
 * 去前缀：provides 作者写裸名 name；canonical_name 由 SDK 派生
 * <app_id>.<name>。caller 侧 requires / callCapability 写全名、不经此派生（不对称）。
 */
export function canonical(appId: string, name: string): string {
  return `${appId}.${name}`;
}
